tools/loader/ImageClassifier.cpp

```diff
@@ -430,25 +446,50 @@ int main(int argc, char **argv) {
   llvm::outs() << "Model: " << loader.getFunction()->getName() << "\n";
 
   int numErrors = 0;
+
+  // Image tensors loaded up to be run at once for benchmark mode.
+  std::vector<std::unique_ptr<Tensor>> preLoadedImages;
+  std::vector<std::unique_ptr<ExecutionContext>> contexts;
+
   while ((streamInputFilenamesMode &&
           getNextImageFilenames(&inputImageBatchFilenames)) ||
          (miniBatchMode &&
           getNextMiniBatch(inputImageBatchFilenames, inputImageFilenames,
                            minibatchIndex, miniBatch)) ||
          isFirstRun) {
-    // Load and process the image data into the inputImageData Tensor.
-    loadImagesAndPreprocess(inputImageBatchFilenames, &inputImageData,
-                            imageNormMode, imageChannelOrder, imageLayout);
-
+    auto inputImageData = llvm::make_unique<Tensor>();
+    if (iterationsOpt) {
```

Comment:
I'm really bothered that it adds so much code to this already-complicated while loop just to support "iterations".  In my mind adding iterations should be a two line diff that just adds a loop :-).  Instead there's kind of a whole new way of doing everything and that feels brittle :-/

Reply:
A fair concern, I considered bypassing the while loop entirely but I think that would add more complexity, since it does seem to be almost a separate flow entirely maybe I do bypass the while loop entirely?

Reply:
I guess I wonder why we can't just `inputImageBatch` like we do in the non-iterations case.

Reply:
I'm not sure I follow, inputImageBatch is the number of images for a single request. We'd still need the logic to run multiple requests right?

Reply:
Maybe the branch should be above this loop then? It would be nice to refactor the loading logic into a helper to avoid this loop getting hard to follow. Generally I find code hard to read if a block's begin and end don't fit on the same screen.